src/CMakeLists.txt

```diff
@@ -69,55 +69,73 @@ if (HAVE_MMAN_H AND (GCRYPT_FOUND OR NSS_FOUND))
 endif()
 
 add_library(openscap SHARED ${OBJECTS_TO_LINK_AGAINST})
+add_library(openscap_testing SHARED ${OBJECTS_TO_LINK_AGAINST})
 
 # default symbol visibility is hidden
 # exported symbols must be marked by OSCAP_API
 set_target_properties(openscap PROPERTIES
 	VERSION ${SONAME}
 	SOVERSION ${SOVERSION}
-	C_VISIBILITY_PRESET hidden
 )
 
-target_link_libraries(openscap ${LIBXML2_LIBRARIES} ${LIBXSLT_LIBRARIES} ${LIBXSLT_EXSLT_LIBRARIES} ${PCRE_LIBRARIES} ${CURL_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+if(CMAKE_C_COMPILER_ID STREQUAL "GNU" OR CMAKE_C_COMPILER_ID STREQUAL "Clang")
+	if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+		set(VERSION_SCRIPT "${CMAKE_CURRENT_BINARY_DIR}/libopenscap.map")
+		add_custom_command(
+			OUTPUT "${VERSION_SCRIPT}"
+			COMMAND "${PYTHON_EXECUTABLE}" "${CMAKE_CURRENT_SOURCE_DIR}/generate_export_map.py" --linux --output "${VERSION_SCRIPT}" "${CMAKE_CURRENT_SOURCE_DIR}"
+			COMMENT "Generating export map"
+		)
+		set_target_properties(openscap PROPERTIES
+			LINK_FLAGS "-Wl,--version-script=\"${VERSION_SCRIPT}\""
+		)
+	endif()
+	add_custom_target(export_map DEPENDS "${VERSION_SCRIPT}")
```

Comment:
I think that this should be moved into the if now and moved back after it is merged in a separate PR.